src/codeql.ts

```diff
@@ -49,29 +51,33 @@ libraryPathDependencies: codeql-${language}`
   );
   fs.writeFileSync(queryFile, query);
 
-  await exec.exec(codeql, [
+  const execBqrs = exec.exec(codeql, [
     "query",
     "run",
     `--database=${database}`,
     `--output=${bqrs}`,
     queryFile,
   ]);
 
-  await exec.exec(codeql, [
+  const execCSV = exec.exec(codeql, [
     "bqrs",
     "decode",
     "--format=csv",
     `--output=${path.join("results", "results.csv")}`,
     bqrs,
   ]);
-  await exec.exec(codeql, [
+
+  const execJSON = exec.exec(codeql, [
     "bqrs",
     "decode",
     "--format=json",
     `--output=${json}`,
     "--entities=all",
     bqrs,
   ]);
+
+  await Promise.all([execBqrs, execCSV, execJSON]);
```

Comment:
I captured all these promises and then await them at once, so that they are not all blocking each other from starting while awaiting the completion of the previous. If you are worried about the machine's ability to handle these concurrently then perhaps this should be reverted, but generally if you await every async item individually you lose some of the benefits of async.

Reply:
OK, seems that `codeql` process didn't like that. Took me a moment to realise that project didn't auto-build TS before running tests, else I'd have noticed sooner.

Reply:
> ... generally if you await every async item individually you lose some of the benefits of async.

:+1: In this case I really just want to write sequential code, but the library is has an async interface, which makes sense I suppose in case you want that.

Specifically in the `codeql` case, it typically expects not to be invoked concurrently against a single project.

> Took me a moment to realise that project didn't auto-build TS before running tests, else I'd have noticed sooner.

Is there an easy way to do that? Just add it to the `test*` tasks?